Update wxGTK docs

vadz opened the pull request:
See [this ticket](https://trac.wxwidgets.org/ticket/18994) for motivation.

vadz committed:
Remove controls implemented natively from "non-native" list

Both GtkSearchEntry and GtkCalendar are used in wxGTK whenever possible.

vadz committed:
Remove outdated wxGTK overview

This doesn't seem to contain any still relevant information not already
mentioned somewhere else.

vadz committed:
Update and condense wxGTK README

Remove obsolete information and parts already documented elsewhere.

Also remove the part about wxWidgets crashing your harddisk or
destroying your monitor because I'm a killjoy.

vadz committed:
Update and simplify wxGTK installation instructions too

Remove obsolete information, add more useful configure options and a
link to CMake.

oneeyeman1 commented:
@vadz ,
I think we can add the Open Indiana to the list of supported GTK platform.

imciner2 commented on docs/gtk/install.md:
```diff
@@ -119,108 +76,93 @@ Now create your super-application myfoo.cpp and compile anywhere with
----------------------------
 
-Usage:
+Configure options                        {#gtk_options}
```
The configure option for disabling the EGL backend and using the X11 backend instead for the OpenGL canvas should be added to this list.

vadz replied:
Thanks, good idea, added now.

PBfordev commented:
As I wrote elsewhere, I know nothing about Linux so my question is probably stupid and this is not usually done.

In addition to mentioning dependency on gtk development package, would it not be useful to list the major 3rd party dependencies that may not generally be installed?
E.g., `wxMediaCtrl` depends on GStreamer, version at least X.Y, `wxWebView` depends on webkitgtk version at least X.Y, `wxGLCanvas` ...

I understand this can be learnt by carefully inspecting config.log but still...

vadz committed:
Don't use hard TABs in the makefile snippet in the docs

This doesn't pass the indentation style check and TABs are likely to get
lost when copying/pasting anyhow, so remove them and add an explicit
note about TABs being needed here instead.

vadz committed:
Document OpenGL-related configure options in wxGTK docs

They're sufficiently important and not obvious to be mentioned.

vadz committed:
Also document wxMediaCtrl and wxWebView dependencies

These are arguably important enough to be mentioned too.

vadz commented:
Thanks @PBfordev, I've added a couple of big libraries, as you've suggested. This is certainly incomplete, but OTOH I don't know if specifying all the dependencies here is worthwhile as it will almost certainly get out of date pretty quickly -- there are just too many things to keep track of.

PBfordev commented:
I think that listing dependencies for `wxMediaCtrl `and ` wxWebView` could be enough (I guess it would be more complicated with OpenGL). 

I thought that perhaps all other 3rd party dependencies (zip, regex? expat, image formats...) fall back to the built-in versions when their system versions are not found.
